src/node_buffer.cc

```diff
@@ -1182,8 +1207,12 @@ void Swap16(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
   SPREAD_ARG(args.This(), ts_obj);
 
-  for (size_t i = 0; i < ts_obj_length; i += 2) {
-    SWAP_BYTES(ts_obj_data, i, i + 1);
+  CHECK_EQ(ts_obj_length % 2, 0);
+  uint16_t* data16 = reinterpret_cast<uint16_t*>(ts_obj_data);
```

Comment:
Looking into this, thanks for pointing it out. The code works as-is with your suggested test below (`buf.slice(1, buf.byteLength - 1).swap16()`), but I'm trying to find out if this violates any rules or if it will break on certain architectures.